### ISSUE TITLE
Failed to deserialize "response.failures" array of GET _tasks/<id> API

### ISSUE THREAD

**Issue by h-h-** [elastic/elasticsearch-net#7889]
**Elastic.Clients.Elasticsearch version**: 8.9.1

**Elasticsearch version**: 8.7.1

**.NET runtime version**: .NET FW 4.7.2

**Operating system version**: Win 10 22H2

**Description of the problem including expected versus actual behavior**:
This code leads to an Exception, when the response contains failures (the array is not empty):

```
getTasksResponse = await _client.Tasks.GetAsync(deleteTaskId, _ct);
```
`deleteTaskId` is a string. E.g. "3RjlPV5DRnGpZ1BBcf2ZBw:2888073"
`_ct` is a cancellation token

Exception:
```
System.Text.Json.JsonException
The JSON value could not be converted to System.String. Path: $.response.failures[0] | LineNumber: 0 | BytePositionInLine: 835.
   bei System.Text.Json.ThrowHelper.ReThrowWithPath(ReadStack& state, Utf8JsonReader& reader, Exception ex)
   bei System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   bei System.Text.Json.JsonSerializer.ContinueDeserialize[TValue](ReadBufferState& bufferState, JsonReaderState& jsonReaderState, ReadStack& readStack, JsonTypeInfo jsonTypeInfo)
   bei System.Text.Json.JsonSerializer.<ReadFromStreamAsync>d__76`1.MoveNext()
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei System.Threading.Tasks.ValueTask`1.get_Result()
   bei Elastic.Transport.DefaultResponseBuilder`1.<SetBodyAsync>d__12`1.MoveNext() in /home/runner/work/elastic-transport-net/elastic-transport-net/src/Elastic.Transport/Components/Pipeline/DefaultResponseBuilder.cs:Zeile 285.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei Elastic.Transport.DefaultResponseBuilder`1.<ToResponseAsync>d__5`1.MoveNext()
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei Elastic.Transport.HttpWebRequestTransportClient.<RequestAsync>d__7`1.MoveNext() in /home/runner/work/elastic-transport-net/elastic-transport-net/src/Elastic.Transport/Components/TransportClient/HttpWebRequestTransportClient.cs:Zeile 179.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei Elastic.Transport.DefaultRequestPipeline`1.<CallProductEndpointAsync>d__55`1.MoveNext() in /home/runner/work/elastic-transport-net/elastic-transport-net/src/Elastic.Transport/Components/Pipeline/DefaultRequestPipeline.cs:Zeile 271.
--- Ende der Stapelüberwachung vom vorhergehenden Ort, an dem die Ausnahme ausgelöst wurde ---
   bei System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   bei System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   bei Elastic.Transport.DefaultHttpTransport`1.<RequestAsync>d__17`1.MoveNext() in /home/runner/work/elastic-transport-net/elastic-transport-net/src/Elastic.Transport/DefaultHttpTransport.cs:Zeile 252.

System.InvalidOperationException
Cannot get the value of a token type 'StartObject' as a string.
   bei System.Text.Json.ThrowHelper.ThrowInvalidOperationException_ExpectedString(JsonTokenType tokenType)
   bei System.Text.Json.Utf8JsonReader.GetString()
   bei System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   bei System.Text.Json.Serialization.JsonCollectionConverter`2.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, TCollection& value)
   bei System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   bei System.Text.Json.Serialization.Metadata.JsonPropertyInfo`1.ReadJsonAndSetMember(Object obj, ReadStack& state, Utf8JsonReader& reader)
   bei System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   bei System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   bei System.Text.Json.Serialization.Metadata.JsonPropertyInfo`1.ReadJsonAndSetMember(Object obj, ReadStack& state, Utf8JsonReader& reader)
   bei System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   bei System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   bei System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
```

Example response:
```
{
  "completed": true,
  "task": {
    "node": "3RjlPV5DRnGpZ1BBcf2ZBw",
    "id": 17158,
    "type": "transport",
    "action": "indices:data/write/delete/byquery",
    "status": {
      "total": 5,
      "updated": 0,
      "created": 0,
      "deleted": 0,
      "batches": 1,
      "version_conflicts": 5,
      "noops": 0,
      "retries": {
        "bulk": 0,
        "search": 0
      },
      "throttled_millis": 0,
      "requests_per_second": -1.0,
      "throttled_until_millis": 0
    },
    "description": "delete-by-query [abc]",
    "start_time_in_millis": 1691737487784,
    "running_time_in_nanos": 143998429,
    "cancellable": true,
    "cancelled": false,
    "headers": {}
  },
  "response": {
    "took": 141,
    "timed_out": false,
    "total": 5,
    "updated": 0,
    "created": 0,
    "deleted": 0,
    "batches": 1,
    "version_conflicts": 5,
    "noops": 0,
    "retries": {
      "bulk": 0,
      "search": 0
    },
    "throttled": "0s",
    "throttled_millis": 0,
    "requests_per_second": -1.0,
    "throttled_until": "0s",
    "throttled_until_millis": 0,
    "failures": [{
        "index": "abc",
        "id": "2000_0000000_",
        "cause": {
          "type": "version_conflict_engine_exception",
          "reason": "[2000_0000000_]: version conflict, required seqNo [80], primary term [1]. current document has seqNo [85] and primary term [3]",
          "index_uuid": "7QzcQMmWTEuAR0TYlsRSYg",
          "shard": "0",
          "index": "abc"
        },
        "status": 409
      }, {
        "index": "abc",
        "id": "2000_0000001_",
        "cause": {
          "type": "version_conflict_engine_exception",
          "reason": "[2000_0000001_]: version conflict, required seqNo [81], primary term [1]. current document has seqNo [86] and primary term [3]",
          "index_uuid": "7QzcQMmWTEuAR0TYlsRSYg",
          "shard": "0",
          "index": "abc"
        },
        "status": 409
      }, {
        "index": "abc",
        "id": "2000_0000002_",
        "cause": {
          "type": "version_conflict_engine_exception",
          "reason": "[2000_0000002_]: version conflict, required seqNo [82], primary term [1]. current document has seqNo [87] and primary term [3]",
          "index_uuid": "7QzcQMmWTEuAR0TYlsRSYg",
          "shard": "0",
          "index": "abc"
        },
        "status": 409
      }, {
        "index": "abc",
        "id": "2000_0000003_",
        "cause": {
          "type": "version_conflict_engine_exception",
          "reason": "[2000_0000003_]: version conflict, required seqNo [83], primary term [1]. current document has seqNo [88] and primary term [3]",
          "index_uuid": "7QzcQMmWTEuAR0TYlsRSYg",
          "shard": "0",
          "index": "abc"
        },
        "status": 409
      }, {
        "index": "abc",
        "id": "2000_0000004_",
        "cause": {
          "type": "version_conflict_engine_exception",
          "reason": "[2000_0000004_]: version conflict, required seqNo [84], primary term [1]. current document has seqNo [89] and primary term [3]",
          "index_uuid": "7QzcQMmWTEuAR0TYlsRSYg",
          "shard": "0",
          "index": "abc"
        },
        "status": 409
      }
    ]
  }
}
```

**Steps to reproduce**:
 1. Update documents during a running delete_by_query request
 2. Check state of the delete_by_query task

**Expected behavior**
Correct deserialization

**Comment by h-h-** [comment]
When to expect any progress on this?

**Comment by flobernd** [comment]
Hi @h-h- , did you try this again with the latest version of the client?

The tasks API was fixed in the specification and the changes have been applied in this PR:
https://github.com/elastic/elasticsearch-net/pull/7920/files#diff-cd8184017cdeb70e7550b9687f5661add4e156eadbf3596e65266da3858d6a31L36

The `response` property of the `GetTaskStatusResponse` class got changed from `Elastic.Clients.Elasticsearch.Tasks.TaskStatus` to `object`. This should fix your deserialization issue.

The reasons for this change are described here:
https://github.com/elastic/elasticsearch-specification/pull/2250

Please let me know, if everything works for you with the 8.10 client.

**Comment by h-h-** [comment]
I can confirm now that the issue was fixed with **Elastic.Clients.Elasticsearch 8.9.3** and is still fixed in 8.11.0.
